### DISCUSSION TITLE
Read only schema

### DISCUSSION THREAD

**Discussion by S7012MY** [prisma/prisma#19512] [Q&A] [answered]
I'm running 2 separate apps in the same database, but different schemas, and only one of them uses prisma.
Now I want to read the other schema inside prisma, but I want to be 100% sure that no new data will be written.

I could create a new db user for this, but I think migrations would be generated anyway, and I don't want this.

Is there a way to do this?

Thank you

**Comment by nurul3101** [comment]
Hello @S7012MY 👋 

You are looking for a Prisma Read Only Client which you can achieve through [Prisma Client Extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions).
Here's an example that you can use:
- https://github.com/prisma/prisma-client-extensions/tree/main/readonly-client

**Reply by S7012MY** [reply]
Thanks a lot! It looks like what I need. Is there a way to use it with prisma studio?

**Reply by nurul3101** [reply]
Do you mean you want to only allow reads through Prisma Studio and disable updates/writes?

**Comment by janpio** [comment, marked as answer]
Did I understand correctly that you want 1 Prisma schema file, that contains models for tables from 2 different database schemas, to be used in the same app, but you do not want to be able to modify data in the models of schema #2 and also ensure that no migrations touch the tables from schema #2?

If so, I don't think you can easily make Prisma work this way yet. We only have feature requests to ignore some tables for Migrations and Introspection - but nothing as granular as I understood your feature. Via `multiSchema` you can generate a Prisma Client that can talk to both database schemas, but then Prisma Client and Migrations will apply to all tables. (And Prisma Studio can not be configured for granular model permissions at all yet.)

Your best approach is probably to create a database user that has read/write on database schema #1, and only read (also no ability to modify the tables) on database schema #2. Then if you do not modify the (introspected or hand-written) Prisma schema of the tables in database schema #2 you will not get any migrations generated, and if you make a mistake at least they will not be able to be executed. Via the Client Extension you can block write or edits to the models from database schema #2 as well - and if that fails (or you prefer not to do that) then the user will not have rights to do anything there either.

**Reply by nurul3101** [reply]
@S7012MY Did you have a chance to look at Jan's recommendation? Let us know in case you still need any help.

**Reply by S7012MY** [reply]
Thanks a lot, this is great
